ingest/scripts/extract-genbank-annotations.py

```diff
@@ -0,0 +1,332 @@
+# appearance date is unknown, or the date of sample receipt by the lab
+# if collection date is unknown), encoded as the "epiweek" followed by
+# a `.`, followed by a 2-digit year. Empirically, sometimes the
+# epiweek value is absent and only a year is present; additionally,
+# sometimes the year is given as a 4-digit year instead of 2.
+# Pragmatically, year values less than 30 will be assumed to be in
+# the 2000s; values greater than 30 will be assumed to be in the
+# 1900s.
+#
+# If there are multiple samples that are not distinguished by the
+# above rules, an additional monotonically-increasing integer is
+# appended after the trailing `/` character. Note that, when present,
+# this additional disambiguation character is NOT followed by an
+# additional `/` delimiter.
+#
+# The WHO genotype designation, if present, will be found after all
+# the above fields, in square brackets (e.g., `[2B]`)
+#
+# There may also be a trailing designation, usually in parentheses,
+# indicating the sample arises from an individual with congenital
+# rubella syndrome (CRS), newborns with a congenital rubella infection
+# (CRI), or a vaccine-derived strain (VAC). For cases marked with CRS
+# or CRI, the geographic localization is the place of birth and onset
+# date of disease is the date of birth.
+#
+# Samples may also have a "collection_date" annotation in the "source"
+# feature. If that is present and is not just a 4-digit year, this
+# script will output that for the date field; otherwise, if a genotype
+# with an epiweek and a year is present, the script will parse the
+# epiweek and year into a year-month date with an ambiguous day
+# component (e.g., 2020-01-XX for a epidate of `01.20`). If both
+# "collection_date" and the epi-date consist of only a year, the
+# script will output the "collection_date" year.
+#
+# In cases where the script can detect something that looks like a
+# genotype in the sample name, it will extract and report that. The
+# genotypes reported are limited to those defined by WHO (i.e., 1a,
+# 1B-1J, 2A-2C); they are further normalize to the set of provisional
+# and accepted genotypes given in the [2013 WHO Rubella nomenclature
+# publication][], so a 1A genotype will be normalized to 1a; a 1j
+# genotype will be normalized to 1J.
+#
+# [2013 WHO Rubella nomenclature publication]: https://www.who.int/publications/i/item/WER8832
+#
+# Occasionally, neither the strain and isolation annotations will
+# contain anything that can be parsed as an "offical" WHO genotype,
+# but there will be a `/note` annotation that looks like "genotype 1a"
+# or "Genotype: 2B". In that case the script will extract and return
+# the genotype, normalized as described above.
+
+
+def main():
+    args = _parse_args()
+
+    tsv_writer = csv.writer(sys.stdout, delimiter="\t")
+    tsv_writer.writerow(
+        [
+            "accession",
+            "date",
+            "genbank_genotype",
+        ]
```

Comment:
Just did another pass focusing on the final ingest output. I was surprised that 70% of the records did not have a `strain` and then realized that it wasn't included in the output from GenBank. 

Thoughts on including `strain` in the output so that it can be merged with the metadata? Then we can see where `date` and `genbank_genotype` was parsed from without having to dig into the GenBank file.

Reply:
> Thoughts on including `strain` in the output so that it can be merged with the metadata? Then we can see where `date` and `genbank_genotype` was parsed from without having to dig into the GenBank file.

yeah, that's probably a good idea.